lib/mongo/error/no_server_available.rb

```diff
@@ -29,11 +29,30 @@ class NoServerAvailable < Error
       #   satisfied.
       #
       # @since 2.0.0
-      def initialize(server_selector)
+      def initialize(server_selector, labels = nil)
+        @labels = labels || []
         super("No server is available matching preference: #{server_selector.inspect} " +
                 "using server_selection_timeout=#{server_selector.server_selection_timeout} " +
                 "and local_threshold=#{server_selector.local_threshold}")
       end
+
+      # Does the error have the given label?
+      #
+      # @example
+      #   error.label?(label)
+      #
+      # @return [ true, false ] Whether the error has the given label.
+      #
+      # @since 2.6.0
+      def label?(label)
```

Comment:
`has_label?`, analogous to e.g. `has_key?` on Hashes.

Reply:
rubocop suggests not using `has_` as a prefix on boolean methods like this, specifically citing that [`key?`](https://ruby-doc.org/core-2.5.1/Hash.html#method-i-key-3F) is preferred to `has_key?`. I don't feel strongly on this though, so I can change this if you prefer.

Reply:
Let's keep it as `label?`.